docs/bare_metal/c_small_libs.md

```diff
@@ -22,6 +22,29 @@ You can build with the smaller C libraries by creating an `mbed_app.json` with t
 
 This links your application with `microlib` for the `ARM` toolchain and `newlib-nano` for the `GCC_ARM` toolchain.
 
+### Non-returning main() required
+
+Exiting from `main()` is not supported by [Arm microlib](#arm-microlib) and causes a bare metal application to crash. Here we show two ways to prevent this.
```

Comment:
I don't see why someone would want to return from main() in a microcontroller especially if the is no OS.

I think the reason we are not explaining the lack of argument passed to main() is also reason enough not mention returning from main().

Reply:
The only case I can think of is when an application does everything in interrupts instead of `main()`. @jenia81: Could you confirm this was your case?

Reply:
It is useful to add this information which was raised as an issue.